wordpress-mobile/WordPress-Android
# Plans on jp dashboard card

Adds dashboard card for wpcom plans

Fixes #18419 

<img width=320 src="https://github.com/wordpress-mobile/WordPress-Android/assets/990349/a48b00b3-3714-4c2c-9918-104142bf5ea1" />


To test:

1. Launch Jetpack app
2. Ensure previous [domain card](https://github.com/wordpress-mobile/WordPress-Android/pull/18286 ) is shown
3. Go to **App Settings** (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
4. Tap **Debug Settings**
5. Find **dashboard_card_free_to_paid_plans** [under Remote features](https://github.com/wordpress-mobile/WordPress-Android/pull/18418)
6. Enable the flag, and restart the app (scroll down)
7. Verify **plans card** appears as shown above on dashboard
8. Verify **domain card** doesn't appear
9. Tap on overflow (three dots) menu
10. **Verify** it pops up menu with **Hide this** option
11. Tap on Hide this
12. **Verify**, the card gets hidden 

>** Note **
>Copy on the card could change later

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests and manual tests

3. What automated tests I added (or what prevented me from doing so)
Updated and added unit tests

PR submission checklist:
